lib/cloudant.js

```diff
@@ -524,21 +531,52 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
           model, where, data, options);
   var self = this;
   var mo = self.selectModel(model);
+  var idName = self.idName(model);
+  // if idName exists, change the idName to be the default `_id` key
+  if (idName && data[idName]) {
+    data._id = data[idName];
+    delete data[idName];
+  }
+  var res = {};
+  var docList = [];
+
   self.all(model, {where: where}, {raw: true}, function(err, docs) {
     if (err) return cb(err, docs);
     if (docs.length === 0) return cb(null, {count: 0});
 
     data = getPlainJSONData.call(self, model, data);
+
+    /**
+     * for each of the docs from the given filter, only update the instance where the revs match
+     * behavior of bulk update in cloudant:
```

Comment:
```
// Change all the names!
var data = [
 {_id: '1', '_rev':'1-123456', name: 'Sakib'},
 {_id: '2', '_rev': '1-432423', name: 'Janny'},
 {_id: '3', '_rev': '1-612412', name: 'Kevin'},
];

var docs = [
 {_id: '1', '_rev':'1-123456', name: 'Bob'},
 {_id: '2', '_rev': '1-432423', name: 'Ted'},
 {_id: '3', '_rev': '1-612412', name: 'Joe'},
];

_.shuffle(docs); // Maybe the docs aren't in the same order when we get them!

_.mergeWith(docs, data, function(dest, src) { return src; });
```

When we run the update, are we sure that the values from data are being merged together with their corresponding items in docs?
If we're deleting `_id` out of them, how do we match them up to do the merge? Even if we don't do that, where's the logic that's using the `_id` field to determine which objects to zip together?